bundles/persistence/org.openhab.persistence.influxdb/java/org/openhab/persistence/influxdb/internal/InfluxDBPersistenceService.java

```diff
@@ -415,24 +421,31 @@ private Object convertBigDecimalToNum(BigDecimal value) {
   /**
    * Converts {@link State} to objects fitting into influxdb values.
    * 
-   * @param state to be converted
-   * @return integer or double value for DecimalType and PercentType, an integer for DateTimeType
-   *         and 0 or 1 for OnOffType and OpenClosedType.
+   * @param 	state to be converted
+   * @return 	integer or double value for DecimalType and PercentType, 
+   * 			0 or 1 for OnOffType and OpenClosedType,
+   *         	SortedMap<String, PrimitiveType> for HSBType,
+   *         	integer for DateTimeType,
+   *         	SortedMap<String, PrimitiveType> for PointType
    */
   private Object stateToObject(State state) {
```

Comment:
stateToObject is called from line 192. You should use getStateAs for DimmerItem/RollershutterItem and ColorItem
have a look at https://github.com/openhab/openhab/blob/master/bundles/persistence/org.openhab.persistence.mysql/java/org/openhab/persistence/mysql/internal/MysqlPersistenceService.java#L306